redfish-core/lib/log_services.hpp

```diff
@@ -6326,5 +6368,923 @@ inline void requestRoutesFullAuditLogDownload(App& app)
+        (dbusObjPath.parent_path().filename().find("dcm") != std::string::npos))
+    {
+        isolatedHwItemId = std::format("{}-{}",
+                                       dbusObjPath.parent_path().filename(),
+                                       dbusObjPath.filename());
+    }
+    else
+    {
+        isolatedHwItemId = dbusObjPath.filename();
+    }
+    return isolatedHwItemId;
+}
+
+/**
+ * @brief API used to get redfish uri of the given dbus object and fill into
+ *        "OriginOfCondition" property of LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ *
+ * @return The redfish response with "OriginOfCondition" property of
+ *         LogEntry schema if success else return the error
+ */
+inline void getRedfishUriByDbusObjPath(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const sdbusplus::message::object_path& dbusObjPath,
+    const size_t entryJsonIdx)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, dbusObjPath,
+         entryJsonIdx](const boost::system::error_code& ec,
+                       const dbus::utility::MapperGetObject& objType) {
+        if (ec || objType.empty())
+        {
+            BMCWEB_LOG_ERROR(
+                "DBUS response error [{} : {}] when tried to get the RedfishURI of isolated hareware: {}",
+                ec.value(), ec.message(), dbusObjPath.str);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        RedfishUriListType::const_iterator redfishUriIt;
+        for (const auto& service : objType)
+        {
+            for (const auto& interface : service.second)
+            {
+                redfishUriIt = redfishUriList.find(interface);
+                if (redfishUriIt != redfishUriList.end())
+                {
+                    // Found the Redfish URI of the isolated hardware unit.
+                    break;
+                }
+            }
+            if (redfishUriIt != redfishUriList.end())
+            {
+                // No need to check in the next service interface list
+                break;
+            }
+        }
+
+        if (redfishUriIt == redfishUriList.end())
+        {
+            BMCWEB_LOG_ERROR(
+                "The object[{}] interface is not found in the Redfish URI list. Please add the respective D-Bus interface name",
+                dbusObjPath.str);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        // Fill the isolated hardware object id along with the Redfish URI
+        std::string redfishUri = redfishUriIt->second + "/" +
+                                 getIsolatedHwItemId(dbusObjPath);
+
```

Comment:
As it is URI, this would be suggested to do like
```
        std::string redfishUri = boost::urls::format("{}/{}", redfishUriIt->second, getIsolatedHwItemId(dbusObjPath);
```